spec/predicates/human-review-vcs.md

```diff
@@ -0,0 +1,100 @@
+# Predicate type: VCS Code Reviews
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/vcs/v0.1
+
+Version: 0.1.0
+
+## Purpose
+
+This attestation type is used to describe code review approvals of the simplest
+kind. The general expectation is that reviews are tied to specific changes in a
+version control system (VCS) such as Git.
+
+## Use Cases
+
+Changes to source code are reviewed by other members of the development team.
+The review may be performed using systems like Gerrit or via reviewing
+mechanisms built into version control forges such as GitHub and GitLab.
+
+## Prerequisite
+
+Understanding of the
+[in-toto attestation specification](https://github.com/in-toto/attestation).
+
+## Model
+
+Typically, software development processes incorporate some type of code review.
+Other members of the development team are expected to study the changes
+proposed and approve them, confirming they meet various expectations pertaining
+to code quality, functionality, and so on. Therefore, the general expectation
+is that this attestation is generated as a confirmation, an approval of some
+specific change, and is performed by developers in the supply chain who have
+code review responsibilities.
+
+As this attestation is generated by a version control service or code review
+service, the signatures are from the service rather than the specific users
+issuing the review.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
+        {
+            "name": "<REPO>",
+            "digest": { "gitCommit": "<HASH>" },
+            "uri": "<LINK TO REPO>"
+        },
+    ],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewer": ["<STRING>", ...],
+        "reviewLink": "<LINK TO REVIEW>",
+        "timestamp": "<TIMESTAMP>"
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the
+[in-toto Attestation Framework's parsing rules](../v1/README.md#parsing-rules).
+
+### Fields
+
+`reviewer` _array of strings_, _required_
+
+Indicates the identity of one or more reviewers. MUST NOT be used over the
+attestation's signature to validate the source of the attestation. This field
+supports scenarios where the attestation is issued and signed by the VCS system.
+
+`reviewLink` _URI_, _optional_
+
+Contains a link to the full review. Useful to point to information that cannot
+be captured in the attestation.
+
+`timestamp` _Timestamp_, _optional_
+
+Indicates time of review creation.
+
+## Example
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [{
+        "name": "in-toto-python",
+        "digest": {
+            "gitCommit": "b0bd8ecab5607e174fa403e002c74a666e7edd51"
+        },
+        "uri": "https://github.com/in-toto/in-toto"
+    }],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewer": ["https://github.com/lukpueh"],
```

Comment:
I would be a little concerned with the fact that this reviewer URL isn't entirely immutable, which is why I think I favor the slightly more detailed Crev format for the reviewer field.

Reply:
Could you elaborate? Do you mean because GitHub usernames can change and / or be registered by other people in future? To address that, we could use the user ID instead in the example and specify inline that it must be immutable for a user.

Reply:
I think I've addressed this by including immutable user IDs.